src/checkers/inference/model/serialization/ToStringSerializer.java

```diff
@@ -50,12 +50,9 @@ public int getIndent() {
     public String serializeSlots(Iterable<Slot> slots, String delimiter) {
         List<String> slotStrings = new ArrayList<>();
 
-        boolean first = true;
         for (Slot slot : slots) {
-            String constraintString = first ? "" : delimiter;
-            constraintString += slot.serialize(this);
+            String constraintString = delimiter + slot.serialize(this);
```

Comment:
I agree with this comment and that is the change we had discussed on Friday.
The new commit fixes this.